### PR TITLE
added data validation and global exception handler

### PR DESCRIPTION
**Pull request by BlueVioletTeti**

### CODE REVIEW

File: src/main/java/mate/academy/bookstore/dto/CreateBookRequestDto.java [hunk 1]
```diff
@@ -1,13 +1,20 @@
...
 import lombok.Data;
 
 @Data
 public class CreateBookRequestDto {
+    @NotNull
     private String title;
+    @NotNull
     private String author;
+    @NotNull
     private String isbn;
```
Comment (OlegTuryk) [inline src/main/java/mate/academy/bookstore/dto/CreateBookRequestDto.java:24]: you can use ISBN annotation

**Review by OlegTuryk — APPROVED** [review]

File: pom.xml [hunk 2]
```diff
@@ -71,6 +71,12 @@
 			<artifactId>liquibase-maven-plugin</artifactId>
 			<version>4.23.1</version>
 		</dependency>
+		<dependency>
+			<groupId>org.hibernate.validator</groupId>
```
Comment (vovasalyha) [inline pom.xml]: In Spring Boot it's better to choose starters over concrete dependencies, so I suggest to use validation starter

File: src/main/java/mate/academy/bookstore/dto/CreateBookRequestDto.java [hunk 3]
```diff
@@ -1,13 +1,20 @@
...
 
+import jakarta.validation.constraints.Min;
+import jakarta.validation.constraints.NotNull;
 import java.math.BigDecimal;
 import lombok.Data;
 
 @Data
 public class CreateBookRequestDto {
+    @NotNull
     private String title;
```
Comment (vovasalyha) [inline src/main/java/mate/academy/bookstore/dto/CreateBookRequestDto.java:16]: Validate the size of title to not allow empty titles and to not exceed the maximum size set by DB schema, the same for author field

File: src/main/java/mate/academy/bookstore/dto/CreateBookRequestDto.java (hunk 1 above)
Comment (vovasalyha) [inline src/main/java/mate/academy/bookstore/dto/CreateBookRequestDto.java:24]: isbn has other validation rule except of not null (see Oleg's suggestion)

File: src/main/java/mate/academy/bookstore/exceptions/CustomGlobalExceptionHandler.java [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+    @Override
+    protected ResponseEntity<Object> handleMethodArgumentNotValid(
+            MethodArgumentNotValidException ex,
+            HttpHeaders headers,
+            HttpStatusCode status,
+            WebRequest request
+    ) {
+        Map<String, Object> body = new LinkedHashMap<>();
+        body.put("timestamp", LocalDateTime.now());
+        body.put("status", HttpStatus.BAD_REQUEST);
```
Comment (vovasalyha) [inline src/main/java/mate/academy/bookstore/exceptions/CustomGlobalExceptionHandler.java]: Usually invalid arguments it's 422 response code. Also, define the common response format for the application and create a class with exception response body.

**Review by vovasalyha — CHANGES_REQUESTED** [review]
GJ. Let's improve it a bit

**Review by nacenik — APPROVED** [review]
LGTM! 👍

**Pull request merged**